src/trip-repo.js

```diff
@@ -3,51 +3,95 @@ import Trip from "./trip.js";
 class TripRepo {
   constructor(tripData, destinationData) {
     this.allTrips = tripData;
-    this.destinationData = destinationData;;
+    this.destinationData = destinationData;
   }
 
   compareDates(d1, d2) {
-    const date1 = new Date(d1)
+    const date1 = new Date(d1);
     const date2 = new Date(d2);
-
     return date1 > date2 ? true : false;
   }
 
+  matchDestinationNames(tripArray) {
+    const linkDestination = tripArray.map(trip => {
+      const findDestination = this.destinationData.find(dest => dest.id === trip.destinationID);
+      const newTrip = new Trip(trip, findDestination);
+      return newTrip;
+    });
+    return linkDestination;
+  }
+
+  createOnlyDateDestination(tripArray, userID) {
+    const currentUserTrips = tripArray.filter(trip => trip.userID === userID);
+    const destinationArray = this.matchDestinationNames(currentUserTrips);
+    const newObject = destinationArray.map(trip => ({'date':trip.date, 'destination':trip.destination.destination}));
+    return newObject;
+  }
+
   findUserPastTrips(userID, date) {
-    const goodDates = [];
-    const dateDestinationArray = [];
+    // QUESTION FOR HUGH: I want to return the element because those are the objects I need but when I return it it comes back as False/True. Is this because my matchDestionNames returns only true/false? I also tried to make this chunk of code into its own function but flippting the date and element.date got really funky...
```

Comment:
So, when using `Array.filter` it will _always_ return a boolean rather than the element itself, which is why your `console.log` on line `36` is probably printing `true/false`. However, this would still result in a filtered array of elements from the original array.

One thing I'm seeing below it that you're passing `this.allTrips.filter` to `this.compareDates` which doesn't seem right. I think you probably want to rewrite lines `34-39` as something like the below:

```javascript
const pastTrips = this.allTrips.filter(trip => {
  if (this.compareDates(date, element.date)) {
    return element;
  }
});
```

If you want to get _really_ fancy you could write it as:

```javascript
const pastTrips = this.allTrips.filter(trip => this.compareDates(date, trip.date));
```

As for what you mentioned about flipping the dates and chunking this out to it's own function, let's talk about its at our next sync-up or you can Slack me with more specifics.

Reply:
OH MY, thank you so much Hugh!!!!!